### PR TITLE
Rewrite tests to not depend on currently broken `dblp` dataset

### PR DESCRIPTION
**Pull request by Padarn**
This PR changes tests that were using the `dblp` as it seems to be currently broken.

### CODE REVIEW

File: test/loader/test_hgt_loader.py [hunk 1]
```diff
@@ -175,13 +175,3 @@ def forward(self, x, edge_index, edge_weight):
     out2 = hetero_model(hetero_batch.x_dict, hetero_batch.edge_index_dict,
                         hetero_batch.edge_weight_dict)['paper'][:batch_size]
     assert torch.allclose(out1, out2, atol=1e-6)
-
-
-@withPackage('torch_sparse>=0.6.15')
```
Comment (Padarn) [inline test/loader/test_hgt_loader.py:180]: Couldn't see what this was testing that wasn't tested by adding the above line, so have removed it

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Added `BaseStorage.get()` functionality ([#5240](https://github.com/pyg-team/pytorch_geometric/pull/5240))
 - Added a test to confirm that `to_hetero` works with `SparseTensor` ([#5222](https://github.com/pyg-team/pytorch_geometric/pull/5222))
 ### Changed
+- Changed tests relying on `dblp` datasets to instead se  synthetic data. ([#5250](https://github.com/pyg-team/pytorch_geometric/pull/5250))
```
Comment (rusty1s) [inline CHANGELOG.md:14]:
```suggestion
- Changed tests relying on `dblp` datasets to instead use  synthetic data. ([#5250](https://github.com/pyg-team/pytorch_geometric/pull/5250))
```

File: test/data/test_lightning_datamodule.py [hunk 3]
```diff
@@ -274,9 +281,17 @@ def test_lightning_hetero_node_data(get_dataset):
 @onlyFullTest
 @withPackage('pytorch_lightning')
 def test_lightning_hetero_link_data(get_dataset):
```
Comment (rusty1s) [inline test/data/test_lightning_datamodule.py:283]:
```suggestion
def test_lightning_hetero_link_data():
```

File: test/loader/test_hgt_loader.py (hunk 1 above)
Comment (rusty1s) [reply test/loader/test_hgt_loader.py:180]: It tests that `HGTLoader` always returns the full set of node and edge types. We can disable this test for via `pytest.skipif` until the dataset is fixed.

Comment (Padarn) [reply test/loader/test_hgt_loader.py:180]: Okay. I've also added the same test in the above
```assert set(batch.edge_types) == set(data.edge_types)```
but makes sense to keep it

Comment (Padarn) [reply test/loader/test_hgt_loader.py:180]: I added just a skip - it didn't seem worth giving a condition. I'll instead open an issue now and note  this as a todo? WDTY?

**Review by rusty1s — APPROVED** [review]

File: test/loader/test_hgt_loader.py (hunk 1 above)
Comment (rusty1s) [reply test/loader/test_hgt_loader.py:180]: Yeah, that‘s what I meant. Thanks!

Comment (Padarn) [reply test/loader/test_hgt_loader.py:180]: Done here https://github.com/pyg-team/pytorch_geometric/issues/5251

**Pull request merged**